lighthouse-cli/test/smokehouse/smoke-test-dfns.js

```diff
@@ -5,129 +5,162 @@
  */
 'use strict';
 
+const assert = require('assert');
 const path = require('path');
-const smokehouseDir = 'lighthouse-cli/test/smokehouse/';
 
-/** @type {Array<Smokehouse.TestDfn>} */
-const SMOKE_TEST_DFNS = [{
+/**
+ * @param {string} filePath
+ */
+function pathRelativeToProjectRoot(filePath) {
+  const projectRoot = path.join(__dirname, '../../../');
+  return path.relative(projectRoot, filePath);
+}
+
+/**
+ * Attempt to resolve a path relative to the smokehouse folder.
+ * If this fails, attempts to locate the path
+ * relative to the project root.
+ * @param {string} payloadPath
+ * @return {string}
+ */
+function resolveLocalOrProjectRoot(payloadPath) {
+  let resolved;
+  try {
+    resolved = require.resolve(__dirname + '/' + payloadPath);
+  } catch (e) {
+    const cwdPath = path.resolve(__dirname + '/../../../', payloadPath);
+    resolved = require.resolve(cwdPath);
+  }
+
+  return resolved;
+}
+
+/** @type {Array<Smokehouse.Test>} */
+const smokeTests = [{
   id: 'a11y',
-  config: smokehouseDir + 'a11y/a11y-config.js',
-  expectations: 'a11y/expectations.js',
+  expectations: require('./a11y/expectations.js'),
```

Comment:
static require paths for the smoke tests I'm all for, but the new `{path, value}` structure of config seems unnecessarily confusing. would it be possible to move to the model in #8962 of colocating the config/expectations and staticly require'ing in *that* file? IIUC that should alleviate the need for path here since we move to the transient config files :)

Reply:
Let's drop the path/value thing in favor of writing temporary files for the config when passing to the smokehouse runner. That interface was the only reason path is needed.

Colocating the expectation/config seems like a bigger change to me and not one that's needed for static resolving to work.